node/src/testing/test_block_builder.rs

```diff
@@ -0,0 +1,194 @@
+use std::collections::BTreeMap;
+
+use casper_hashing::Digest;
+use casper_types::{
+    system::auction::BLOCK_REWARD, EraId, ProtocolVersion, PublicKey, SecretKey, Timestamp, U512,
+};
+use rand::Rng;
+
+use crate::{
+    components::consensus::EraReport,
+    types::{Block, BlockHash, BlockPayload, Deploy, DeployHashWithApprovals, FinalizedBlock},
+};
+
+const MAX_ERA_FOR_RANDOM_BLOCK: u64 = 6;
+
+pub(crate) struct TestBlockBuilder {
```

Comment:
I'd have probably put this in a `block_builder` module inside the `block` module, just to avoid this `testing` module becoming a dumping ground for all test builder helpers.  Not a big deal though.